src/client/Sidebar/Item.jsx

```diff
@@ -1,35 +1,76 @@
 import { useState, useCallback } from 'react';
 
 export const Item = ({
+  name,
   children,
   handleClick,
   handleRenameClick,
   handleDeleteClick,
+  isDir,
 }) => {
   const [isHovered, setIsHovered] = useState(false);
 
+  const [isRenaming, setIsRenaming] = useState(false);
+
+  const [renameValue, setRenameValue] = useState('');
+
+  const onKeyDown = (event) => {
+    if (event.key === 'Enter' || event.key === 'Escape') {
+      event.target.blur();
+    }
+  };
+  const onBlur = (event) => {
+    if (event.target.value.trim() === '') {
+      setEditingValue(value);
+    } else {
+      setIsRenaming(false);
+      handleRenameClick(event.target.value);
+    }
+  };
+
   const handleMouseEnter = useCallback(() => {
     setIsHovered(true);
   }, []);
 
   const handleMouseLeave = useCallback(() => {
     setIsHovered(false);
+    setIsRenaming(false);
   }, []);
 
   return (
     <div
       className="full-box file-or-directory"
-      onClick={handleClick}
+      onClick={
+        isRenaming ? console.log(isRenaming) : handleClick
+      }
       onMouseEnter={handleMouseEnter}
       onMouseLeave={handleMouseLeave}
     >
-      <div className="name">{children}</div>
+      <div className="name">
+        {isRenaming ? (
+          <input
+            type="text"
+            aria-label="Field name"
+            value={renameValue}
+            onKeyDown={onKeyDown}
+            onBlur={onBlur}
+            onChange={(event) =>
```

Comment:
Best to move this to a separate definition outside the JSX and use `useCallback`.